packages/blade/src/components/InfoGroup/InfoGroup.web.tsx

```diff
@@ -0,0 +1,466 @@
+          {children}
+        </BaseBox>
+      </BaseBox>
+    );
+  },
+);
+
+const _InfoItem = (
+  { children, testID, isHighlighted }: InfoItemProps,
+  ref: React.Ref<BladeElementRef>,
+): ReactElement => {
+  const { itemOrientation, isHighlighted: contextIsHighlighted } = useInfoGroup();
+  const [hasAvatar, setHasAvatar] = React.useState(false);
+  const isVertical = itemOrientation === 'vertical';
+  const shouldHighlight = isHighlighted ?? contextIsHighlighted;
+
+  const infoItemContextValue = React.useMemo(
+    () => ({
+      hasAvatar,
+      setHasAvatar,
+      isHighlighted: shouldHighlight,
+    }),
+    [hasAvatar, shouldHighlight],
+  );
+
+  if (isVertical) {
+    return (
+      <InfoItemContext.Provider value={infoItemContextValue}>
+        <FlexItemBox ref={ref as never} testID={testID} isHighlighted={shouldHighlight}>
+          {children}
+        </FlexItemBox>
+      </InfoItemContext.Provider>
+    );
+  }
+
+  return (
+    <InfoItemContext.Provider value={infoItemContextValue}>
+      <ContentsItemBox ref={ref as never} testID={testID}>
+        {children}
+      </ContentsItemBox>
+    </InfoItemContext.Provider>
+  );
+};
+
+/**
+ * InfoItem
+ *
+ * Container component that wraps InfoItemKey and InfoItemValue pairs.
+ * Manages the layout and visual connection between key-value elements.
+ *
+ * ----
+ *
+ * #### Usage
+ *
+ * ```tsx
+ * <InfoItem>
+ *   <InfoItemKey>Account Holder</InfoItemKey>
+ *   <InfoItemValue>Saurabh Daware</InfoItemValue>
+ * </InfoItem>
+ * ```
+ *
+ * ----
+ * Checkout {@link https://blade.razorpay.com/?path=/docs/components-infogroup InfoGroup Documentation}
+ */
+const InfoItem = assignWithoutSideEffects(React.forwardRef(_InfoItem), {
+  displayName: 'InfoItem',
+  componentId: 'InfoItem',
+});
+
+// InfoGroup Component
+const _InfoGroup = (
+  {
+    children,
+    itemOrientation = 'horizontal',
+    size = 'medium',
+    keyAlign = 'left',
+    valueAlign = 'left',
+    isHighlighted = false,
+    gridTemplateColumns,
+    width,
+    maxWidth,
+    minWidth,
+    testID,
+    paddingLeft,
+    paddingRight,
+    paddingTop,
+    paddingBottom,
+    padding,
+    paddingX,
+    paddingY,
+    ...rest
+  }: InfoGroupProps,
+  ref: React.Ref<BladeElementRef>,
+): ReactElement => {
+  const contextValue = React.useMemo(
+    () => ({
+      size,
+      itemOrientation,
+      keyAlign,
+      valueAlign,
+      isHighlighted,
+    }),
+    [size, itemOrientation, keyAlign, valueAlign, isHighlighted],
+  );
+
+  const defaultGridTemplateColumns =
+    itemOrientation === 'horizontal'
+      ? 'max-content 1fr'
+      : `repeat(min(4, ${React.Children.count(children)}), 1fr)`;
+
+  // Use provided gridTemplateColumns or default based on itemOrientation
+  const templateColumns = gridTemplateColumns ?? defaultGridTemplateColumns;
+
+  return (
+    <InfoGroupContext.Provider value={contextValue}>
+      <BaseBox
+        ref={ref as never}
+        as="dl"
```

Comment:
Do we have other related `as` props for `dt` and `dd`?? If not we should either remove this and use normal spans or add related tags in other components.

Reply:
yes yes I've added the dt and dd as well. Following the same structure as MDN example: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dd